### DISCUSSION TITLE
Workspace groups

### DISCUSSION THREAD

**Discussion by waterkip** [i3/i3#5646] [Q&A] [answered]
Is something like this possible with i3?
The premise is that I want to have a couple of workspaces that share a similar layout. I call these workspace groups, for lack of a better word. Each workspace group contain three actual workspaces: dev/debug/cli. They all have a layout, and when used for the first time, all the applications are fired up for you. I want to have workspace 1/2/3 to be always dev, debug and cli. These workspaces differ depending on the group (clients/company/private/etc.). When I hit a shortcut, more or less the following action happens:

```
to_group=$1

# Get the current group name from i3, which is stored somewhere in i3
# TODO: Is there such a mechanism?
group=$(i3-msg get-variable activeWorkspaceGroup)
i3-msg "workspace dev; rename workspace to $group-dev"
i3-msg "workspace debug; rename workspace to $group-debug"
i3-msg "workspace cli; rename workspace to $group-cli"

WORKSPACES=($(i3-msg -t get_workspaces | jq -r '.[] | .name'))
for i in dev debug cli
do
   echo $WORKSPACES | grep -q $to_group-$i
   if [ $? -ne 0 ]
   then
      # It does not exist. Apply layout and start all apps
      i3-msg "workspace $i; append_layout /path/to/layout.json"
      i3-msg "workspace $i; exec $terminal"
      # rinse repeat
   else
      i3-msg "workspace $to_group-$i; rename workspace to $i"
   fi
done
i3-msg "set-variable activeWorkspaceGroup $to_group"
```

Can one set and get variables via some i3 command? i3-msg isn't able to do so.

**Comment by okraits** [comment]
Did you have a look at https://i3wm.org/docs/layout-saving.html ?

i3-msg doesn't know the variables in the i3 config. The i3 config variables are replaced in the file while parsing. But you can use shell variables with i3-msg.

**Reply by waterkip** [reply]
See the `i3-msg "workspace $i; append_layout /path/to/layout.json` bit in my code example :) So yes, I am aware of layout saving.

**Comment by waterkip** [comment, marked as answer]
I wrote something about the solution I wrote a while ago:

https://wesley.schwengle.net/article/workspace-on-demand-in-i3wm-84a4/
